File: Kata 7/Tidy Number/Tidy Number (Special Numbers Series #9).py
def tidyNumber(n):
    #pass  
    arr=list(str(n))
    for i in range (1,len(arr)):
        if int(arr[i])-int(arr[i-1]) <0:
            return False
    return True
	
	
#判斷是否所有後面的值大於前面	
def tidyNumber(n): 
	arr=list(str(n))
	return all (int(arr[i])- int(arr[i-1]) >=0 for i in range (1,len(arr))) 
	
#判斷排序完的值是否與位排序一樣
def tidyNumber(n):
    s = list(str(n))
    return s == sorted(set(s))
	
def tidyNumber(n):    
    return str(n) == ''.join(sorted(list(str(n))))
	

File: Kata 7/Tidy Number/Tidy Number (Special Numbers Series #9).cs
using System.Linq;
class Kata
{
	public static bool tidyNumber (int number)
    {
          // Your Code is Here .... Enjoy !!
            char []s  = number.ToString().ToCharArray();

            for (int i = 1; i < s.Length; i++)
            {
                if (Convert.ToInt32(s[i].ToString()) - Convert.ToInt32(s[i - 1].ToString()) < 0)
                {
                    return false;
                }
            }

            return true;
    }
	public static bool tidyNumber(int number)
    {
       return string.Join("", number.ToString().ToCharArray().Select(x => Convert.ToInt32(x.ToString())).OrderBy(y=>y) )== number.ToString();
    }
	
	public static bool tidyNumber(int number)
    {
       int i = 0;
       var q = number.ToString().ToCharArray();           
       return q.Skip(1).All(x => x - q[i++]>=0);
    }  
   
}